src/ai/defaultai.cc

```diff
@@ -4843,13 +5008,12 @@ bool DefaultAI::check_productionsites(const Time& gametime) {
 			return false;
 		}
 
-		const uint32_t trees_in_vicinity = map.find_immovables(
-		   game(), Widelands::Area<Widelands::FCoords>(map.get_fcoords(site.site->get_position()), 5),
-		   nullptr,
-		   Widelands::FindImmovableAttribute(Widelands::MapObjectDescr::get_attribute_id("tree")));
+		unsigned trees_nearby = find_immovables_nearby(site.site->descr().collected_attributes(),
```

Comment:
as this is dealing with rangers we need to find created immovables rather then collected ones.

Reply:
same here as with lumberjacks even when using created attributes the code delivers different numbers as in trunk. which results in rangers not stopped even if totally surrrounded with trees.

Reply:
Ok I found the problem we can't count collected Immovables from a ranger as he has none. We can't count created immovables neither as they are saplings so I made a solution to count the collected attributes from the suppoprted buildings which delivered the correct attributes and numbers. Will upload my changes again tonight.